app/main.go

```diff
@@ -613,25 +619,40 @@ func makeSpamLogWriter(opts options) (accessLog io.WriteCloser, err error) {
 	}, nil
 }
 
-func setupLog(dbg bool, secrets ...string) {
-	logOpts := []lgr.Option{lgr.Msec, lgr.LevelBraces, lgr.StackTraceOnError}
-	if dbg {
-		logOpts = []lgr.Option{lgr.Debug, lgr.CallerFile, lgr.CallerFunc, lgr.Msec, lgr.LevelBraces, lgr.StackTraceOnError}
-	}
+func setupLog(dbg bool, loggingFormat string, secrets ...string) error {
 
-	colorizer := lgr.Mapper{
-		ErrorFunc:  func(s string) string { return color.New(color.FgHiRed).Sprint(s) },
-		WarnFunc:   func(s string) string { return color.New(color.FgRed).Sprint(s) },
-		InfoFunc:   func(s string) string { return color.New(color.FgYellow).Sprint(s) },
-		DebugFunc:  func(s string) string { return color.New(color.FgWhite).Sprint(s) },
-		CallerFunc: func(s string) string { return color.New(color.FgBlue).Sprint(s) },
-		TimeFunc:   func(s string) string { return color.New(color.FgCyan).Sprint(s) },
-	}
-	logOpts = append(logOpts, lgr.Map(colorizer))
+	var logger *slog.Logger
+	if loggingFormat == "json" {
```

Comment:
My brain is going to explode by those nested else/ifs ;) It would be nice to simplify it and make it as plain as possible. Maybe a top-level switch with an early exit will help.

Reply:
fixed